src/poetry/core/packages/utils/link.py

```diff
@@ -49,7 +49,7 @@ def __str__(self) -> str:
             return str(self.url)
 
     def __repr__(self) -> str:
-        return "<Link %s>" % self
+        return f"<Link {self}>"
```

Comment:
```suggestion
        return f"<Link {self!s}>"
```

To keep this code consistent (`%s` implies the same `__str__` but not `__format__` call).